---
title: "**Family Markers: Using Multiply-Affected Families to Identify Risk Genes**"
format:
    html: default 
    pdf: default
---

This module teaches about ...

## Learning Objectives

1. Interpret pedigree information to determine the suitability of a DNA marker for tracking a disease trait in a family (GSA)[https://genetics-gsa.org/education/genetics-learning-framework/].

2. Build a pedigree using bioinformatics tools (R).

## Materials

You will need an internet connection for this activity as written. Students can use a local installation of R or RStudio if the data is downloaded ahead of time.

Download this module as:

-   Quarto (qmd)
-   Rmarkdown (Rmd)
-   Word (docx)

Download the R student activity as:

-   Quarto (qmd)
-   Rmarkdown (Rmd)
-   Word (docx)

Borrow images from [Google Slides](https://docs.google.com/presentation/d/1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII/edit?usp=sharing).

## Activity Part 1

### Background

First, let's read the "plain language" summary from a publication entitled "Rare variants implicate NMDA receptor signaling and cerebellar gene networks in risk for bipolar disorder".

> This research explores the genetic basis of bipolar disorder, a serious mental health condition where people experience extreme mood swings between mania and depression. The researchers studied DNA from 41 families with multiple members affected by bipolar disorder. They identified 741 genes potentially linked to bipolar disorder, which overlapped with genes known to be involved in other brain development disorders. Their most significant finding was a specific variant in a gene called DAO (D-amino acid oxidase). When they tested this variant in human cells, it reduced the amount and activity of the DAO protein. In mice with the equivalent genetic change, they observed: (1) less DAO protein in certain brain regions, (2) greater susceptibility to stress, (3) altered responses to drugs affecting NMDAR (a type of brain receptor), and (4) changes in gene activity patterns in the cerebellum (a brain region). The team also found similar gene expression patterns in the cerebellum of human bipolar disorder patients. The study suggests bipolar disorder may involve disruptions to NMDAR signaling and gene expression in the cerebellum, providing new insights into how this condition develops at the molecular level.

::: {.callout-note}
## Check Your Knowledge

1. What does the DAO gene do?

2. How did researchers identify DAO and other possible genes linked to bipolar disorder?
:::

### Start Posit Cloud

In the next steps, we will construct a pedegree from the publiation described above.

We will use Posit Cloud for this activity. Go to <https://posit.cloud/plans/free> and follow the steps to sign up. You can also Log In using the button on the top right if you have used Posit Cloud in the past.

<!-- Callout about what cloud computing is -->

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII/edit?slide=id.g1965a5f7f0a_0_44#slide=id.g1965a5f7f0a_0_44")
```

Once logged in, select "New Project" and "New RStudio Project". It will take a few seconds to deploy and load.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII/edit?slide=id.g357ffc10808_0_3#slide=id.g357ffc10808_0_3")
```

Rename your project "Family Markers" by clicking on "Untitled Project".

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII/edit?slide=id.g357ffc10808_0_12#slide=id.g357ffc10808_0_12")
```

### Package Install and Load

We will need to install the `kinship2` package for this activity.

Type the following into your console and press return to run the code. You will see a lot of red text, but usually that's a good thing!

```{r eval=F}
install.packages("kinship2")
```

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII/edit?slide=id.g357ffc10808_0_30#slide=id.g357ffc10808_0_30")
```

Next, we will load the package so it's ready to use:

```{r eval=F}
library(kinship2)
```

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII/edit?slide=id.g357ffc10808_0_61#slide=id.g357ffc10808_0_61")
```

::: {.callout-tip}
## What are R Packages?

Packages are collections of R code, data, and documentation that extend the base functionality of R. Think of them like "expansion packs" on top of your basic R software.

Packages are developed by the R community and made available through repositories like CRAN (Comprehensive R Archive Network), Bioconductor, and GitHub. They are especially useful if you want to do a specialized kind of analysis, such as genomic analysis!

We use the `library` command to load and attach packages to the R environment. This means links the package you downloaded to your current session of R.
:::

### Get Pedigree Data

We'll read in some data from the GEMs website. Run the following code in your console:

```{r}
ped_data <- read.csv("https://genomicseducation.org/data/pedigree_data.csv")
ped_data
```

If everything went correctly, you should see both an object called `ped_data` in the Environment Pane (top right) and the data printed to the console.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII/edit?slide=id.g357ffc10808_0_47#slide=id.g357ffc10808_0_47")
```

::: {.callout-tip}
## What is an R object?

An object is something stored in a programming language, such as data. Think of it like a "noun". 

In R, we assign objects using the `<-` symbol, with the name of your choice on the left, and the contents of the object on the right. 

Objects will show up in your Environment Pane. It's a good idea to check the Environment Pane regularly.
:::

Enter the following code in the console to view the data in tabular format:

```{r eval=F}
View(ped_data)
```

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII/edit?slide=id.g357ffc10808_0_74#slide=id.g357ffc10808_0_74")
```

## Questions Part 1

::: {.callout-note}
## Check Your Knowledge

1. Why might we want to install add-on packages in R?

2. How can we confirm that our data loaded correctly?

3. Assume the column "bipolar" is 1=yes, 0=no, and NA=missing. Assume the column "depression" is the same. Based on this information, what do you think the column "diagnosis_available" represents?
:::

## Activity Part 2

...
