---
title: "Genomic Medicine and Polygenic Risk Scores"
format:
    html: default
---

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_6#slide=id.g35f4636d8f5_0_6")
```

## Part 1

In Part 1, we'll go over the background and data for this activity.

Many human diseases are caused by a combination of genetic and environmental factors. For most of these diseases, it can be difficult to predict what combination will lead to a person developing disease. Instead, physicians and researchers talk about the increased risk - essentially, we know what factors increase the likelihood of developing diseases, even if we can't predict with 100% certainty whether someone might get sick. 

### Polygenic Inheritance

When we think about genetically-inherited diseases, we usually think about those where a single gene variant can cause disease. Most of the classic examples of genetic disease, like sickle cell anemia, Tay Sachs, or Huntington's, are passed along via _Mendelian inheritance_. A mutation in a single gene is enough to disrupt normal protein synthesis and cellular processes, causing a person to become sick. 

<!-- Image of one allele being different -->

However, most diseases, particularly common diseases, are not the result of a mutation to a single gene. Instead, they are the result of mutations to many different genes. Each mutation itself isn't enough to cause problems, but when all of them show up together, the cellular processes aren't able to function normally.  This type of inheritance (when a trait is caused by a combination of mutations to many genes) is called _polygenic inheritance_. Polygenic traits are quite common, and they aren't always diseases! Any trait that shows a range of phenotypes is likely to be the result of polygenic inheritance. Some of the more famous polygenic traits include height and skin color.

### Polygenic Risk Scores

Researchers have developed something called a “polygenic risk score” (PRS) to estimate how likely a person might be to develop a disease based on their genetics.

::: {.callout-tip}
It’s always important to remember that **having a higher PRS does not guarantee someone will get a disease.**  Likewise, having a lower PRS does not guarantee someone is protected from a disease. 

A PRS is just a rough estimate based on what we know at the moment about the genetics that contribute to developing a disease. In many case, a high PRS will only increase a person's overall risk for a disease by less than 5%.
:::

At its core, the polygenic risk score is a simple idea. If multiple variants can contribute to the development of a disease, other variants might also provide protection from a disease. We can look at all the variants a person might have, add them up, and then get an idea of whether they have more variants that contribute to a disease relative to how many protective variants they have.

Being able to calculate a PRS requires that scientists have an understanding of all the genes and noncoding regions of DNA that might contribute to developing (or not developing) a disease. **A PRS is only as good as the reference database for a particular disease or trait.**

### Genomic Ancestry

<!-- More here? -->

### Exploring Variant Data

In the next steps, we'll be looking at how PRS can help patients be more informed about disease risk. 

This example will look at prostate carcinoma, or prostate cancer. While prostate cancer is common in men and is a leading cause of cancer-related death, it tends to be slow growing with limited aggressiveness (see [https://www.ncbi.nlm.nih.gov/books/NBK470550/ ]( https://www.ncbi.nlm.nih.gov/books/NBK470550/ )). This means that genetic screening and symptom monitoring, especially in older age, can have a big impact on outcomes. 

### Mr. J's Data

We're going to take gene screening data from an imaginary patient, Mr. J, to understand his risk. Mr. J has African ancestry, which will be important later.

Get the data at [https://genomicseducation.org/data/prs_ind_1.txt](https://genomicseducation.org/data/prs_ind_1.txt). It should look like:

```
rs7463326:G
rs58235267:G
rs74001374:C
...
```

::: {.callout-tip}
## What does the data mean?

Each line of the data represents a variant location in the genome and individual person's version of that gene (allele). 

For example, **rs7463326** is the variant. Mr. J has a "G" here. Other patients might have "A" or "C".
:::

### Exploring Variants

Let's explore the **rs7463326** variant a bit more. Go to [https://www.ebi.ac.uk/gwas/variants](https://www.ebi.ac.uk/gwas/variants) and type "rs7463326" in the search bar.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f43ab35e5_0_6#slide=id.g35f43ab35e5_0_6")
```

Select the first result.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f43ab35e5_0_11#slide=id.g35f43ab35e5_0_11")
```

Notice the variant information. The most severe **rs7463326** variant lies in an [intergenic region](https://www.genome.gov/genetics-glossary/Intergenic-Regions#:~:text=Definition,Medical%20Genetics%20Branch), or between protein coding genes. 

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_37#slide=id.g35f4636d8f5_0_37")
```

Scroll down to see the risk allele associated with the variant. For **rs7463326**, the risk allele is "G" and it's associated with prostate carcinoma.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_42#slide=id.g35f4636d8f5_0_42")
```

Mr. J has a "G" here. Other patients might have "A" or "C". This means he might be more at risk than other patients. However, we know risk is often dictated by multiple genes. In the next steps we'll explore how to assess risk with multiple variants.

## Part 1 Questions

::: {.callout-note}
## Check Your Knowledge

1. When treating genetic disease, why is it important to examine multiple genes, rather than examine one gene at a time?

1. Which of these are accurate statements?

    a. Having certain gene variants can mean increased risk of disease.  
    b. Higher PRS scores guarantee development of disease.
    c. PRS must be based on an appropriate database of gene variants.
    d. All variants come from protien-coding genes.

1. Try looking up another variant at [https://www.ebi.ac.uk/gwas/variants](https://www.ebi.ac.uk/gwas/variants). Look up "rs2075650". What kind of variant is listed under "Most severe consequence"? Are there any diseases associated with this variant?

:::

## Part 2

In Part 2, we'll calculate our first PRS score.

### Entering Data

Go to https://prs.byu.edu/calculate_score.html - this is the website we'll use to calculate a PRS for Mr. J's data.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f43ab35e5_0_0#slide=id.g35f43ab35e5_0_0")
```

Return to the data [here](https://genomicseducation.org/data/prs_ind_1.txt). Copy this data into the top part of the PRS calculator website.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_57#slide=id.g35f4636d8f5_0_57")
```

### Selecting the Study

Next, scroll down to "GWAS Summary Statistics". For item #1, "Select Disease(s) or Trait(s) of Interest", type in "prostate carcinoma" and select the checkbox. You might need to type slowly as the dropdown list has a lot of options! Then, click on "Apply Filters" to update the studies that can be used to calculate the PRS.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_63#slide=id.g35f4636d8f5_0_63")
```

Under "Select from Filtered Studies", type in the study number, "GCST011047", or find it from the dropdown menu.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_70#slide=id.g35f4636d8f5_0_70")
```

### Selecting the Population

Select "African" as the "Preferred Super Population" and "1000 Genomes - AFR population" under "MAF Population".

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_70#slide=id.g35f4636d8f5_0_70")
```

Finally, click "Calculate Risk Scores".

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_84#slide=id.g35f4636d8f5_0_84")
```

### Interpreting the Score

The calculated score for this study will appear in the box at the bottom. You will have to scroll over.

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_84#slide=id.g35f4636d8f5_0_84")
```

::: callout-tip
## Viewing Results

It might be easier to view the results if you (1) download the results and (2) open the results in a tabular data program like Microsoft Excel or Google Sheets.
:::

Let's review a few important results. 

```{r, fig.align='center', echo = FALSE, fig.alt= "TO ADD", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1QxPej8YIbgsToOITRYKLxmGScjo0Eu5ldiKEE-lXvF0/edit?slide=id.g35f4636d8f5_0_99#slide=id.g35f4636d8f5_0_99")
```

- **SNP Overlap**: How many variants in the patient's data were also present in the study.
- **Included SNPs**: How many variants were in the study overall. Combined with "SNP Overlap", this tells us Mr. J's data was comprehensive and included all of the 19 variants described in this study.
- **Polygenic Risk Score**: Risk estimate based on all of Mr. J's variants. Because the score is greater than 1, generally that means some increased risk.
- **Percentile**: Risk described relative to others in the population. A percentile of 100 means that Mr. J has greater risk than almost everyone else! He has more of the disease causing variants compared to others in the population.

## Part 2 Questions

::: {.callout-note}
## Check Your Knowledge

1. In order to be accurate, PRS scores must be calculated based on variants collected from the same ancestral population as the patient. This is because some alleles are more rare in specific human ancestries and can confer different amount of risk.

    Mr. J's variant data was carefully collected based on the variants identified in the GCST011047 study (Conti et al. 2021). Go to the [study page](https://www.ebi.ac.uk/gwas/studies/GCST011047) and look at the description under "Discovery ancestry label". 

    Given that Mr. J has African ancestry, is this study an appropriate match to assess his risk? Why or why not?
    
2. Imagine you were helping Mr. J's providers craft a care plan. Based on Mr. J's results, would you recommend more frequent screening? Why or why not?
:::

## Part 3

In Part 3, we'll compare the previous PRS score to one calculated using a different study.

### Adding a Study




