---
title: "Differential Expression and the Mouse Gut-Brain Axis"
date: "`r format(Sys.time(), '%B %d, %Y')`"
bibliography: ../book.bib
csl: ../resources/ieee-with-url.csl
format:
    html: default
---

```{r, include = FALSE}
ottrpal::set_knitr_image_path()
```

This module introduces you to differential expression analyses using the R programming language.

This activity will walk through differential expression analysis and use bioinformatics tools (R) to understand how gut bacteria can influence the expression of genes in the brain (the gut-brain axis). You will work with real data from a mouse RNA-seq experiment to understand how researchers explore how human disorders might be impacted by gut-brain axis using model organisms.

## Part 1

In Part 1, we'll go over some background context for this activity. Then, we'll get set up on Posit Cloud for the next steps.

The data for this activity includes gene expression data from two different brain regions (striatum and prefrontal cortex) in mice. Mice in this experiment received gut bacteria transplants. Some mice received transplants from humans who have been diagnosed with Autism Spectrum Disorder (ASD), while other mice received transplants from humans who did not have any diagnosis (control). This allowed researchers to control the composition of the gut microbiome in each mouse. Researchers then bred the mice and looked for differences in gene expression in the brains of the offspring.

Researchers are interested in gene expression changes caused by the gut bacteria because of something called the _gut-brain axis_, a proposed connection between the gut microbiome and the brain. In recent years, studies have identified links between neurological symptoms and the presence of intestinal inflammation. There are around 100 million neurons in the human gut, and around 70% of the serotonin (an important neurotransmitter) in the human body is made in the gut. Communication between the microbes in the gut and the brain happens through multiple routes, including immune signaling (signals sent to the brain by immune cells), stimulation of the vagus nerve, and activation of the enteric nervous system (a system of neurons in the intestinal that are often called the "second brain").

Changes that happen in the brain as a result of the gut microbiome are suspected to be responsible for the severity of some symptoms associated with neurodevelopmental and neurodegenerative disorders. A possible link between the gut microbiome and increased symptom severity is actively being studied for Major Depressive Disorder, Autism Spectrum Disorder, fibromyalgia, migraines, Parkinson’s disease, Alzheimer’s disease, and more. 

The samples in this dataset are from mice that _do not_ carry genes associated with any neurodevelopmental or neurodegenerative diseases. Instead, any symptoms or changes in the brain they show are the result of microbes introduced to their gut via transplant. **This dataset gives you the opportunity to study possible effects of the gut-brain axis firsthand.**

::: callout-note
It is important to note that researchers are **not** suggesting that ASD or _any neurodevelopment disorder_ is entirely induced by gut bacteria. There is a **strong genetic component to ASD.** Scientists have known for years that there are **both genetic and environmental components** to the development or severity of **some** ASD symptoms. This research explores one possible environmental component (gut bacteria).
:::

**Expand each section to learn more!**

::: {.callout-caution collapse="true"}
# The Gut-Brain Axis 

The _gut-brain axis_ is the term used for the proposed connection between the gut microbiome and the brain. 

Researchers have identified multiple ways the intestinal microbiota in the gut might remodel the structure of the nervous system. This includes changes in myelination, neurogenesis, neurotransmitters and synaptic plasticity, growth of dendrites, activation of microglia, and changes in the permeability of the blood-brain-barrier. The exact mechanism for these changes is still unknown. 

```{r, fig.align='center', echo = FALSE, fig.alt= "How the gut might alter the nervous system in mice.", out.width = '100%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g36e8af3aaa8_34_0#slide=id.g36e8af3aaa8_34_0")
```

In the figure above, the "up" arrow indicates an increase; this could be increased production of a protein, increased activity of a biological pathway, or a general increase in the presence of a structure. The "down" arrow indicates a decrease.

So why are the neural processes mentioned above important?

**myelination**: the process of forming protein sheaths (called myelin) around the long axons of nerves. This insulates the axon and improves the speed and efficiency of transmission of neural signals. 

**neurogenesis**: the formation of new neurons, particularly in the adult central nervous system. This is important for memory and learning.`

**synaptic plasticity**: the process of strengthening or weakening connections at the synapse of nerves (where the signal is transferred from one nerve to another). This sort of adaptive change is fundamental for memory and learning, but also helps with recovery after injury.

**dendritic growth**: the process by which neurons form new branches (dendrites) and spines (small protrusions on the dendrites). This is a key aspect of synaptic plasticity and is essential for brain development, learning, and memory formation.

**microglia**: immune cells found in the central nervous system. They support central nervous system development and synaptogenesis (formation of new synapses), maintain homeostasis, contribute to an immune response against infectious agents, and are associated with adult neurogenesis, neuroinflammation, degenerative diseases, stroke, trauma and regeneration.

**blood-brain-barrier**: a highly selective semi-permeable barrier located in the central nervous system that separates the bloodstream from the fluid that surrounds the brain. It maintains a  stable environment for neuronal function and regulates passage of molecules between the brain and the rest of the body.
:::

::: {.callout-caution collapse="true"}
# Autism Spectrum Disorder: A Neurodevelopmental Disorder

Although Autism Spectrum Disorder (ASD) can be diagnosed at any age, it is considered neurodevelopmental disorders because symptoms usually show up within the first two years of life. Many of these disorders have overlapping symptoms, including difficulty managing emotions, experiencing intense emotional outbursts, a dysregulated stress response system, difficulty with executive functions (like attention regulation, organization, and planning), and issues with social communication and unspoken social norms. Many of these disorders also have similar biological roots, such as neuroinflammation, genetic variation, and disruptions in systems that rely on dopamine, norepinephrine, and serotonin (which can cause impaired connections between regions of the brain, called white matter connectivity). People with these disorders also show differences in brain regions like the prefrontal cortex and the amygdala compared to people without neurodevelopmental disorders. For example, malfunctioning of the hypothalamic-pituitary-adrenal (HPA) axis is common in neurodevelopmental disorders. The HPA axis regulates the brain and body's response to stress. Many people with neurodevelopmental disorders show an unusually strong reaction to stress and can have difficulty recovering from it.

```{r, fig.align='center', echo = FALSE, fig.alt= "Researchers have identified three major factors that may contribute to neurodevelopmental disorders.", out.width = '100%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g344ce1a0afa_0_0#slide=id.g344ce1a0afa_0_0")
```

*Autism spectrum disorder (ASD)* is a neurological disorder that affects behavioral and social interactions, among other things. Individuals diagnosed with ASD can experience a wide range of symptoms, including differences in social behaviors and communication styles, as well as intellectual disabilities and physical issues like sensory sensitivities or gastrointestinal problems. Some genes that have been suggested as ASD risk genes include SHANK3, NRXN1, CHD8, ARIH2, ZDHHC15, and TRPC6. Individuals with ASD also show changes in both white and gray matter volume, as well as an imbalance in connectivity of neural circuits (there are more connections and stronger connections than typical within local regions of the brain, and both fewer and weaker connections between distant regions of the brain). The neural circuits that are imbalanced connect brain regions like the prefrontal cortex and the amygdala.

```{r, fig.align='center', echo = FALSE, fig.alt= "", out.width = '100%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g3707230d9af_0_0#slide=id.g3707230d9af_0_0")
```
:::

::: {.callout-caution collapse="true"}
# Brain Structure: The Striatum and the Prefrontal Cortex

This dataset includes gene expression data from two different regions in the brain: the striatum and the prefrontal cortex. 

```{r, fig.align='center', echo = FALSE, fig.alt= "Image of the striatum in the human brain.", out.width = '100%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g33d1434cdd1_12_0#slide=id.g33d1434cdd1_12_0")
```

The *striatum* is part of the brain involved in motor control and cognitive tasks like reward processing, decision-making, and social interactions (often called executive functions). It lies deep within the center of the brain and is composed of both gray matter (which can be thought of as the "processing" part of brain tissue) and white matter (which is the brain structure involved in transporting messages); the combination of the gray and white matter give this region of the brain a striped appearance, resulting in the name "striatum". The striatum is involved in both reflexive movement - that is, involuntary movement that happens as an immediate response to a stimulus - and slower, planned movement like walking. In Parkinson's disease, some patients experience degeneration of parts of the striatum, resulting in spastic, uncontrollable movement.

```{r, fig.align='center', echo = FALSE, fig.alt= "Image of the prefrontal cortex in the human brain.", out.width = '100%'}
ottrpal::include_slide("
https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g33d1434cdd1_12_10#slide=id.g33d1434cdd1_12_10")
```

The *prefrontal cortex* is the part of the brain that is primarily in charge of decision making, reasoning, personality, maintaining social appropriateness, and other complex behaviors that fall under the umbrella of executive functions. This can include planning, self-control, and working towards long-term goals. The prefrontal cortex is located in the very front of the brain, just behind your forehead. One of the most famous brain injury patients was Phineas Gage, a railroad worker who survived an iron rod through his forehead. His prefrontal cortex was destroyed in this accident, and doctors noted huge behavioral and personality changes. You can read more about his case [here](https://pmc.ncbi.nlm.nih.gov/articles/PMC1114479/).
:::

::: {.callout-caution collapse="true"}
# The original study

The original study, *Human Gut Microbiota from Autism Spectrum Disorder Induces Behavioral Deficits in Mice*, was published in 2019 [@sharon2019]. Gut microbiota are known to be different between individuals with ASD and individuals who are considered typically-developing. Additionally, some individuals with ASD also experience gastrointestinal symptoms, and their gut microbiota show the greatest difference when compared to the gut microbiota of typically-developing individuals. Some researchers have proposed that gut bacteria can influence some of the symptoms of ASD. The relationship between the intestinal microbiome and the development and function of the human brain is known as the *gut-brain axis.*

In this study, researchers explored whether they could induce ASD-like behaviors in mice by changing their gut microbiome. They transplanted stool (which contains gut bacteria) from humans into germ-free mice. The mice received gut bacteria from either humans diagnosed with ASD or humans with no diagnosis. The researchers discovered that colonization with gut microbiota was enough to induce _some_ ASD-like behaviors in the mice. They also let the mice breed and collected gene expression data from the brains of their offspring to explore whether changing the gut microbiota could result in changed gene expression. In particular, they discovered that the offspring of mice who received gut bacteria from ASD donors showed different gene splicing and expression profiles of certain ASD-relevant genes.


You can read the original research paper [here](https://www.cell.com/cell/fulltext/S0092-8674(19)30502-1).
:::


#### The mouse as a model organism

The mouse is the most commonly-used model organism in laboratory work. In fact, mice and rats make up 95% of the lab animal population, and more than 80% of the research that has been awarded the Nobel Prize for Medicine was done at least in part with mouse models [@hu2024; @FBR2025].

So what makes mice such good model organisms for biomedical research? Well, first, they're economical and relatively easy to keep. Since mice are small, they don't require a huge amount of space or food. They also have fast reproductive cycles, so researchers can study multiple generations within only a few years. Most importantly, though, mice and humans are both mammals and have about 85% of their protein-coding genome in common. As a result, mouse physiology is quite similar to human physiology. The mouse circulatory, reproductive, digestive, hormonal, and nervous systems are frequently used as models to study how humans grow, age, and develop chronic diseases. They are particularly important model organisms for cancer research and neuroscience.

You can find additional information about how the mouse is used in research [here!](https://www.informatics.jax.org/mgihome/other/homepage_IntroMouse.shtml)

::: callout-note
## Check Your Knowledge

1.  What is the gut-brain axis?

2.  Why do you think the researchers looked at gene expression in the prefrontal cortex and the striatum?

3.  Why are mice useful as model organisms for human disease studies?
:::

### Start Posit Cloud

In the next steps, we will explore the data from the publication described above.

We will use Posit Cloud for this activity. Go to <https://posit.cloud/plans/free> and follow the steps to sign up. You can also Log In using the button on the top right if you have used Posit Cloud in the past.

<!-- callout about what cloud computing is ? -->

```{r, fig.align='center', echo = FALSE, fig.alt= "Sign up for a free Posit Cloud account before starting this lab.", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g3b501096c6b_1_0#slide=id.g3b501096c6b_1_0")
```

Once logged in, select "New Project" and "New RStudio Project". It will take a few seconds to deploy and load.

```{r, fig.align='center', echo = FALSE, fig.alt= "Click on New Project and New RStudio Project to open your workspace.", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g3b501096c6b_1_6#slide=id.g3b501096c6b_1_6")
```


Rename your project "Differential Expression Mouse GutBrain" by clicking on "Untitled Project".

```{r, fig.align='center', echo = FALSE, fig.alt= "Change the name of your Posit Cloud workspace by clicking here.", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g3b501096c6b_1_13#slide=id.g3b501096c6b_1_13")
```

### Package Install and Load

We will need to install the `tidyverse` package for this activity.

Type the following into your console and press return to run the code. You will see a lot of red text, but usually that's a good thing!

```{r, fig.align='center', echo = FALSE, fig.alt= "This is where you start typing in order to use the console in Posit Cloud", out.width = '90%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g3b501096c6b_1_155#slide=id.g3b501096c6b_1_155")
```

```{r message=FALSE}
install.packages("tidyverse")
```


Next, we will load the package so it's ready to use:

```{r message=FALSE}
library(tidyverse)
```

::: callout-tip
## What are R Packages?

Packages are collections of R code, data, and documentation that extend the base functionality of R. Think of them like "expansion packs" on top of your basic R software.

Packages are developed by the R community and made available through repositories like CRAN (Comprehensive R Archive Network), Bioconductor, and GitHub. They are especially useful if you want to do a specialized kind of analysis, such as genomic analysis!

We use the `library` command to load and attach packages to the R environment. This means links the package you downloaded to your current session of R.

The "tidyverse" package that you loaded is useful for loading, wrangling, and exploring data.
:::

### Load Differential Expression Data

**Differential expression** refers to the process where different cells within an organism, even those with the same DNA, express different genes, with some genes being expressed or "turned on", while others are "turned off". This expression difference is the basis of cell type diversity in multicellular organisms (like mice and humans). Because some diseases are thought to be the result of improper gene expression, researchers will use differential gene expression experiments to pinpoint the genes responsible for the disease symptoms.

We'll read in some data from the GEMs website. Run the following code in your console:

```{r}
prefrontal <- read_csv("https://genomicseducation.org/data/mouse_gutbrain_de_autismVcontrol_in_prefrontalcortex.csv")
```

You have just loaded differential expression data comparing ASD-transplant mice and control-transplant mice for the prefrontal cortex region of the brain. _Remember, ASD-transplant mice are mice that received gut bacteria from humans diagnosed with ASD, while control-transplant mice received gut bacteria from humans without an ASD diagnosis._

::: callout-tip
## Check yourself

What does the prefrontal cortex region of the brain do?
:::

## Part 2

Let's take a look at the data you loaded into R. Type the following code into the console:

```{r}
head(prefrontal)
```

::: callout-tip
### What do the column names mean?

When working with someone else's data, it can be difficult to figure out what a particular column or variable name means. A **data dictionary** is a key to decode these!

- **gene**: the geneID (see below for more detail). 

- **baseMean**: the average of the normalized count values for this gene, taken over all samples.

- **log2FoldChange**: a measure of the change in gene expression for this gene between the ASD and control mice. A positive value indicates upregulation (increased expression), while a negative Log2FC indicates downregulation (decreased expression).

- **lfcSE**: the standard error calculated for log2FoldChange.

- **stat**: t-value calculated to determine if the log2FoldChange is significantly different from zero.

-**pvalue**: the p-value for the log2FoldChange.

-**padj**: the p-value, adjusted for multiple testing.
:::

In this dataset, genes are identified using their Ensembl Gene ID code. Every gene has an ID that looks something like this:

> ENSMUSG00000000001

**ENS** stands for "Ensembl". Ensembl is a [genome database project](https://ensemblgenomes.org/) managed by the European Bioinformatics Institute. It's one of several databases like this. Others include NIH's National Center for Biotechnology Information (NCBI; the organization that manages GenBank and PubMed) and the University of California, Santa Cruz (UCSC) Genome Browser. When a gene code starts with "ENS", it means you should look up the gene code in the Ensembl database.

**MUS** stands for "Mus", which is the genus for the mouse.

**G** stands for "Gene". When you see "G" in the ID name, you know you are working with a gene. There are also codes for transcripts ("T"), exons ("E"), and proteins ("P").

**00000000001** is the numerical code associated with the gene.

### Ranking the genes by log2FoldChange

The log2FoldChange value gives us an idea of which genes show the greatest differential expression between the ASD mice and the control mice. It is a ratio of how many gene transcripts were found in ASD-type mice compared to control-type mice. 

We can arrange the table based on these values (going from smallest to largest) by copying the following code into your console:

```{r}
prefrontal_ascending <- arrange(prefrontal, log2FoldChange)
head(prefrontal_ascending)
```

The genes at the top of this table have the most negative log2FoldChange values. Negative log2FoldChange values means a **decreased expression** (or, a downregulation in gene expression) in the experimental group versus the control group.

We can use a similar command to look at the genes with the largest positive log2FoldChange. Positive log2FoldChange values indicate genes that are upregulated, or have **increased expression** in the experimental group versus the control group.

```{r}
prefrontal_descending <- arrange(prefrontal, desc(log2FoldChange))
head(prefrontal_descending)
```

### Using the Mouse Genome Database

We have identified the genes (by their geneID) that show the greatest differential expression between ASD and control mice. Now let's find out what those genes do!

The [Mouse Genome Informatics](https://www.informatics.jax.org) database that tracks mouse genes and expression data. A full introduction to everything available through the MGI can be found [here](https://www.informatics.jax.org/mgihome/other/homepage_usingMGI.shtml). We'll reproduce some of it below.

To look up information on a particular gene of interest, choose the "Genes" button.

```{r, fig.align='center', echo = FALSE, fig.alt= "To look up information on a particular gene of interest, choose the “Genes” button.", out.width = '100%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g357f7ed062e_9_0#slide=id.g357f7ed062e_9_0")
```

Next, type the gene ID into the "Search" bar up top. We'll look up "ENSMUSG00000079516".

```{r, fig.align='center', echo = FALSE, fig.alt= "Type the gene ID into the “Search” bar up top.", out.width = '100%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g3412c9eee7c_0_1#slide=id.g3412c9eee7c_0_1")
```

After you type the gene ID into the Search bar and hit enter, you should see a new page with basic information about the gene. Click on the gene symbol (in this example, Reg3a) to get more detailed information.

```{r, fig.align='center', echo = FALSE, fig.alt= "Click on the gene symbol (in this example, Reg3a) to get more detailed information.", out.width = '100%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g346083d4dc0_0_0#slide=id.g346083d4dc0_0_0")
```

::: callout-tip
You might come across some unexpected terms when you search the MGI for your gene ID. In addition to genes, Ensembl gene IDs are also given to "pseudogenes", "putative genes", and "lncRNA".

**pseudogene**: This is a stretch of DNA that looks like a gene but doesn't actually code for any protein products. It's essentially a copy of a gene that contains mutations that prevent translation into a protein product. The mutations can include partial deletions, missing promoters, missing start codons, premature stop codons, frameshift mutations, or missing introns. Any of these are enough to result in a pseudogene.

**putative gene**: This is a DNA segment that is believed to be a gene, but its function and protein product has not been confirmed. They are frequently identified based on the presence of an Open Reading Frame. (An Open Reading Frame, or ORF, is a stretch of DNA between a start codon and a stop codon that can be translated into a protein.) Putative genes are not given names until they become confirmed genes.

**lncRNA**: This stands for "long non-coding RNA". lncRNA is a type of RNA molecule that is transcribed from DNA but does not code for proteins. These RNA molecules are at least 200-500 nucleotides long and play roles in various biological processes, like gene regulation.
:::

On the new page that you open, details about the gene are organized into categories. Down the left-hand side of the page, you will see sections about the chromosomal location, homology, gene ontology, expression data, and more. 

If you scroll down the page, you can also examine the pathways and processes the gene product is involved in under the "gene ontology" section. Gene ontology refers to a common, controlled set of vocabulary and terms that researchers use when categorizing gene and protein function, biological processes, and location within the cell. Reg3a is categorized as a protein involved in carbohydrate derivative binding, signaling receptor activity, and signaling receptor binding. Biologically, the Reg3a protein helps with cell differentiation, cell population proliferation, the immune system, and responding to stimuli.

```{r, fig.align='center', echo = FALSE, fig.alt= "If you continue scrolling down the page, you can also examine the pathways and processes the gene product is involved in under the “gene ontology” section.", out.width = '100%'}
ottrpal::include_slide("https://docs.google.com/presentation/d/1vikcyml6uw-_DaC1yFpS-UWoWNEGxFukMB_eCi42BqY/edit?slide=id.g346083d4dc0_0_11#slide=id.g346083d4dc0_0_11")
```


::: callout-note
## Check Your Knowledge

4.  What gene shows the greatest upregulation (increase in gene expression) in ASD-type mice compared to control mice? Give both the gene ID and the gene name.

5.  What molecular functions does this gene do?

6.  What gene shows the greatest downregulation (decrease in gene expression) in ASD-type mice compared to control mice? Give both the gene ID and the gene name.

7.  What biological processes is this gene involved in?
:::

## Part 3

In part 3, we'll create a list of genes that are differentially expressed in the prefrontal cortex, then examine whether they show the same differential expression pattern in the striatum.

### Creating a gene list

In the previous section, we identified genes with the highest and lowest log2FoldChange. However, this metric isn't always the best indicator of differential expression. The adjusted p-value, or padj, is preferred over log2FoldChange for identifying statistically significant changes in gene expression because padj accounts for multiple testing. This is crucial when analyzing a large number of genes simultaneously, as each additional statistical test increases the risk of identifying a false positive. The adjusted p-value provides a statistical correction that minimizes this risk. log2FoldChange provides information about the magnitude of expression change, while padj indicates its statistical significance.

Let's create a list of all the genes that are differentially expressed between ASD and control mice using the `filter` command. We'll set the padj threshold to 0.05.

```{r}
prefrontal_sig <- filter(prefrontal, padj < 0.05)
head(prefrontal_sig)
dim(prefrontal_sig)
```

### Compare differential gene expression across regions

It might be interesting to look at the expression of a gene across all the possible regions. To do this, we will need to load the differential expression data for the striatum.

```{r}
striatum <- read_csv("https://genomicseducation.org/data/mouse_gutbrain_de_autismVcontrol_in_striatum.csv")
```

Then we'll filter out the gene in which we're interested from each object. Let's take a look at gene ENSMUSG00000079516, which is the reg3a gene we previously looked up on MGI.

```{r}
reg3a_prefrontal <- filter(prefrontal, GeneID == "ENSMUSG00000079516")

reg3a_striatum <- filter(striatum, GeneID == "ENSMUSG00000079516")
```

Finally, take a look at the differential expression of reg3a in each region.

In the prefrontal cortex:

```{r}
reg3a_prefrontal
```

In the striatum:

```{r}
reg3a_striatum
```

::: callout-note
## Check Your Knowledge

8.  How many genes did you keep after filtering out any genes with a padj > 0.05?

9.  How many genes in #8 show increased expression in ASD-type mice?

10.  Did the genes you identified in part 2 (the one with the greatest positive log2FoldChange, as well as the one with the greatest negative log2FoldChange) meet the padj threshold? That is, did you keep these genes in your dataset after filtering?

11.  Is the differential expression pattern of Reg3a the same in the striatum as in the prefrontal cortex? 
:::
