---
title: "Differential Expression and the Mouse Gut-Brain Axis"
---

This module introduces you to differential expression analyses using the R programming language.

This activity will walk through differential expression analysis and use bioinformatics tools (R) to understand how gut bacteria can influence the expression of genes in the brain (the gut-brain axis). You will work with real data from a mouse RNA-seq experiment to understand how researchers explore how human disorders might be impacted by gut-brain axis using model organisms.

## Part 1

::: callout-note
## Check Your Knowledge

1.  What is the gut-brain axis?

2.  Why do you think the researchers looked at gene expression in the prefrontal cortex and the striatum?

3.  Why are mice useful as model organisms for human disease studies?
:::


### Package Install and Load

We will need to install the `tidyverse` package for this activity.

Type the following into your console and press return to run the code. You will see a lot of red text, but usually that's a good thing!

```{r}
install.packages("readr")
install.packages("dplyr")
```

Next, we will load the package so it's ready to use:

```{r message=FALSE}
library(readr)
library(dplyr)
```

We'll read in some data from the GEMs website. Run the following code in your console:

```{r}
prefrontal <- read_csv("https://genomicseducation.org/data/mouse_gutbrain_de_autismVcontrol_in_prefrontalcortex.csv")
```

You have just loaded differential expression data comparing ASD-transplant mice and control-transplant mice for the prefrontal cortex region of the brain.

## Part 2

Let's take a look at the data you loaded into R. Type the following code into the console:

```{r}
head(prefrontal)
```


The log2FoldChange value gives us an idea of which genes show the greatest differential expression between the ASD mice and the control mice. It is a ratio of how many gene transcripts were found in ASD-type mice compared to control-type mice. 

We can arrange the table based on these values (going from smallest to largest) by copying the following code into your console:

```{r}
prefrontal_ascending <- arrange(prefrontal, log2FoldChange)
head(prefrontal_ascending)
```

The genes at the top of this table have the most negative log2FoldChange values. Negative log2FoldChange values means a **decreased expression** (or, a downregulation in gene expression) in the experimental group versus the control group.

We can use a similar command to look at the genes with the largest positive log2FoldChange. Positive log2FoldChange values indicate genes that are upregulated, or have **increased expression** in the experimental group versus the control group.

```{r}
prefrontal_descending <- arrange(prefrontal, desc(log2FoldChange))
head(prefrontal_descending)
```

::: callout-note
## Check Your Knowledge

4.  What gene shows the greatest upregulation (increase in gene expression) in ASD-type mice compared to control mice? Give both the gene ID and the gene name.

5.  What molecular functions does this gene do?

6.  What gene shows the greatest downregulation (decrease in gene expression) in ASD-type mice compared to control mice? Give both the gene ID and the gene name.

7.  What biological processes is this gene involved in?
:::

## Part 3

In part 3, we'll create a list of genes that are differentially expressed in the prefrontal cortex, then examine whether they show the same differential expression pattern in the striatum.


In the previous section, we identified genes with the highest and lowest log2FoldChange. However, this metric isn't always the best indicator of differential expression. The adjusted p-value, or padj, is preferred over log2FoldChange for identifying statistically significant changes in gene expression because padj accounts for multiple testing. This is crucial when analyzing a large number of genes simultaneously, as each additional statistical test increases the risk of identifying a false positive. The adjusted p-value provides a statistical correction that minimizes this risk. log2FoldChange provides information about the magnitude of expression change, while padj indicates its statistical significance.

Let's create a list of all the genes that are differentially expressed between ASD and control mice using the `filter` command. We'll set the padj threshold to 0.05.

```{r}
prefrontal_sig <- filter(prefrontal, padj < 0.05)
head(prefrontal_sig)
dim(prefrontal_sig)
```


It might be interesting to look at the expression of a gene across all the possible regions. To do this, we will need to load the differential expression data for the striatum.

```{r}
striatum <- read_csv("https://genomicseducation.org/data/mouse_gutbrain_de_autismVcontrol_in_striatum.csv")
```

Then we'll filter out the gene in which we're interested from each object. Let's take a look at gene ENSMUSG00000079516, which is the reg3a gene we previously looked up on MGI.

```{r}
reg3a_prefrontal <- filter(prefrontal, GeneID == "ENSMUSG00000079516")

reg3a_striatum <- filter(striatum, GeneID == "ENSMUSG00000079516")
```

Finally, take a look at the differential expression of reg3a in each region.

In the prefrontal cortex:

```{r}
reg3a_prefrontal
```

In the striatum:

```{r}
reg3a_striatum
```

::: callout-note
## Check Your Knowledge

8.  How many genes did you keep after filtering out any genes with a padj > 0.05?

9.  How many genes in #8 show increased expression in ASD-type mice?

10.  Did the genes you identified in part 2 (the one with the greatest positive log2FoldChange, as well as the one with the greatest negative log2FoldChange) meet the padj threshold? That is, did you keep these genes in your dataset after filtering?

11.  Is the differential expression pattern of Reg3a the same in the striatum as in the prefrontal cortex? 
:::

