---
title: "**Differential Expression with RNA-seq**"
output:
  html_document
---

```{r, include = FALSE}
ottrpal::set_knitr_image_path()
```

## Overview

This module introduces students to differential expression analyses using the R programming language. Previous programming experience is helpful but not necessary. Each section introduces an RNA-seq dataset from a published research paper. The datasets have been formatted for student use. All the formatting and processing of the raw data has been done, so students can focus on hypothesis testing.

This module is a companion and expansion to the C-MOOR RNA-seq miniCURE. It can be used as an independent activity or as part of a broader miniCURE experience. The original C-MOOR RNA-seq miniCURE explores gene expression in a *Drosophila* midgut dataset. This module adds additional RNA-seq datasets that students can use, either as the primary activity or as an independent research project.

- Here is the student guide if you want to run this activity as a [stand-alone activity](rnaseq_de_student-guide.qmd)

- Here is the student guide if you want to use this as an [extension of the C-MOOR miniCure]((rnaseq_de_independent_project-guide.qmd))

TODO: Add flowchart

You can find out more about the RNA-seq miniCURE and other C-MOOR activities at the [C-MOOR website](https://www.c-moor.org/). An online guide to the miniCURE is [here](https://science.c-moor.org/miniCURE-RNA-seq/index.html).

**Duration:**

## Learning Objectives

## Materials and Setup

## Scientific Topics

## Outline

# Gut-brain axis: Autism Spectrum Disorder

This dataset includes gene expression data from two different brain regions (striatum and prefrontal cortex) in mice. Mice in this experiment received fecal transplants. Some mice received transplants from humans who have been diagnosed with Autism Spectrum Disorder (ASD), while other mice received transplants from humans who did not have any diagnosis (control). This allowed researchers to control the composition of the gut microbiome in each mouse. Researchers then bred the mice and looked for differences in gene expression in the brains of the offspring.

 [*Gut Microbiota from Human Autism Spectrum Disorder Induces Behavioral Deficits in Mice*] (https://pmc.ncbi.nlm.nih.gov/articles/PMC6993574/pdf/nihms-1528786.pdf).

## What comparisons can students make?

In this dataset, students have the option to look at gene expression in control vs ASD mice, as well as gene expression in both striatum and prefrontal cortex. All the mice were male and sacrificed at the same age (45 days).

Additionally, students can look at gene expression in striatum vs prefrontal cortex in only ASD mice or only control mice. Likewise, students also have the option of looking at gene expression in ASD and control mice, focusing only on striatum or only on prefrontal cortex.


## Download the Datasets

**Comparing gene expression between ASD and control mice**

Both brain regions: <https://genomicseducation.org/data/mouse_autism_deseq_autismVcontrol.csv>

Prefrontal cortex only: <https://genomicseducation.org/data/mouse_autism_deseq_autismVcontrol_in_prefrontalcortex.csv>

Striatum only: <https://genomicseducation.org/data/mouse_autism_deseq_autismVcontrol_in_striatum.csv>

**Comparing gene expression between prefrontal cortex and striatum**

All mice: <https://genomicseducation.org/data/mouse_autism_deseq_tissuetype.csv>

Only ASD mice: <https://genomicseducation.org/data/mouse_autism_deseq_tissuetype_in_ASDmice.csv>

Only control mice: <https://genomicseducation.org/data/mouse_autism_deseq_tissuetype_in_controlmice.csv>
