---
title: "**Differential Expression with RNA-seq**"
output:
  html_document
---

```{r, include = FALSE}
ottrpal::set_knitr_image_path()
```

This module introduces students to differential expression analyses using the R programming language. Previous programming experience is helpful but not necessary. Each section introduces an RNA-seq dataset from a published research paper. The datasets have been formatted for student use. All the formatting and qc of the raw data has been done previously, so students can focus on hypothesis testing.

This module is a companion and expansion to the C-MOOR RNA-seq miniCURE. It can be used as an independent activity or as part of a broader miniCURE experience. The original C-MOOR RNA-seq miniCURE explores gene expression in a _Drosophila_ midgut dataset. This module adds additional RNA-seq datasets that students can use, either as the primary activity or as an independent research project. 

TODO: Add flowchart

You can find out more about the RNA-seq miniCURE and other C-MOOR activities at the [C-MOOR website](https://www.c-moor.org/). An online guide to the miniCURE is [here](https://science.c-moor.org/miniCURE-RNA-seq/index.html).


# Gut-brain axis: Autism Spectrum Disorder

This dataset includes gene expression data from two different brain regions (striatum and prefrontal cortex) in mice. Mice in this experiment received fecal transplants. Some mice received transplants from humans who have been diagnosed with Autism Spectrum Disorder (ASD), while other mice received transplants from humans who did not have any diagnosis (control). This allowed researchers to control the composition of the gut microbiome in each mouse. Researchers then bred the mice and looked for differences in gene expression in the brains of the offspring.

## The original study

The original study, _Gut Microbiota from Human Autism Spectrum Disorder Induces Behavioral Deficits in Mice_, was published in 2019. Gut microbiota are known to be different between individuals with ASD and individuals who are considered typically-developing. Additionally, some individuals with ASD also experience gastrointestinal symptoms, and their gut microbiota show the greatest difference when compared to the gut microbiota of typically-developing individuals. Some researchers have proposed that gut bacteria can influence some of the symptoms of ASD. The relationship between the intestinal microbiome and the development and function of the human brain is known as the _gut-brain axis._

In this study, researchers explored whether they could induce ASD-like behaviors in mice by changing their gut microbiome. They transplanted stool from either humans with ASD or controls into germ-free mice and discovered that colonization with gut microbiota was enough to induce ASD-like behaviors in the mice. They also let the mice breed and collected gene expression data from the brains of their offspring to explore whether changing the gut microbiota could result in changed gene expression. In particular, they discovered that the offspring of mice who received stool from ASD donors showed different gene splicing and expression profiles of certain ASD-relevant genes.

It is important to note that researchers are **not** suggesting that ASD is entirely induced by gut bacteria. There is a strong genetic component to ASD. Scientists have known for years that there are **both genetic and environmental components** to the development or severity of **some** ASD symptoms. This research explores one possible environmental component.

You can read the original research paper [here](https://pmc.ncbi.nlm.nih.gov/articles/PMC6993574/pdf/nihms-1528786.pdf).

## What comparisons can students make?

In this dataset, students have the option to look at gene expression in control vs ASD mice, as well as gene expression in both striatum and prefrontal cortex. All the mice were male and sacrificed at the same age (45 days).

Additionally, students can look at gene expression in striatum vs prefrontal cortex in only ASD mice or only control mice. Likewise, students also have the option of looking at gene expression in ASD and control mice, focusing only on striatum or only on prefrontal cortex.

## What does the student guide include?

The student guide includes some background information that may help your students understand the kinds of questions they may want to ask with this dataset. 

### Study overview and definitions

In addition to the basic overview of the study above, the student guide also includes definitions of Autism Spectrum Disorder and explanations of the parts of the brain sampled in this study. It also includes a link to the [story of Phineas Gage](https://pmc.ncbi.nlm.nih.gov/articles/PMC1114479/), a railroad worker who survived an iron rod through his forehead, and whose injury led to breakthroughs in our understanding of neuroanatomy. 

### The mouse as a model organism

Students may not be familiar with why the mouse is used as a model organism. This section provides a very general explanation of the role of mice in laboratory research, as well as why mice can be used as models for human disease.

### Learning about mouse genes

This section of the student guide includes information on how to navigate the [Mouse Genome Informatics](https://informatics.jax.org/) database maintained by The Jackson Laboratory. It also has a brief description on interpreting a mouse gene ID code and non-genes that have associated IDs.

## Downloads

### Student Guides
A separate student guide and Rmd file (with example code) has been created for this activity. The student guide can be accessed as either a webpage or as a downloadable pdf. The Rmd file should be downloaded. Once it is downloaded, students can open Rmd file with RStudio. This Rmd file can also be used in any cloud-based R environment, such as AnVIL or PositCloud.

You can find the student guide for working through this activity and dataset [here](rnaseq_de_student-guide.qmd)

A separate version of the student guide has been provided if you are using this module as an extension of the C-MOOR miniCURE. In this version, students are given examples of the coding tasks they did in the miniCURE activities, but have been provided empty code blocks for writing their own code. This modified guise is [here](rnaseq_de_independent_project-guide.qmd)

### Datasets

Full gene expression data with no comparisons, just counts (_NOTE:_ This dataset is only used in the independent project guide):  [https://genomicseducation.org/data/mouse_autism_counts.csv](https://genomicseducation.org/data/mouse_autism_counts.csv)

**Comparing gene expression between ASD and control mice**

Both brain regions: [https://genomicseducation.org/data/mouse_autism_deseq_autismVcontrol.csv](https://genomicseducation.org/data/mouse_autism_deseq_autismVcontrol.csv)

Prefrontal cortex only:  [https://genomicseducation.org/data/mouse_autism_deseq_autismVcontrol_in_prefrontalcortex.csv](https://genomicseducation.org/data/mouse_autism_deseq_autismVcontrol_in_prefrontalcortex.csv)

Striatum only: [https://genomicseducation.org/data/mouse_autism_deseq_autismVcontrol_in_striatum.csv](https://genomicseducation.org/data/mouse_autism_deseq_autismVcontrol_in_striatum.csv)

**Comparing gene expression between prefrontal cortex and striatum**

All mice: [https://genomicseducation.org/data/mouse_autism_deseq_tissuetype.csv](https://genomicseducation.org/data/mouse_autism_deseq_tissuetype.csv)

Only ASD mice: [https://genomicseducation.org/data/mouse_autism_deseq_tissuetype_in_ASDmice.csv](https://genomicseducation.org/data/mouse_autism_deseq_tissuetype_in_ASDmice.csv)

Only control mice: [https://genomicseducation.org/data/mouse_autism_deseq_tissuetype_in_controlmice.csv](https://genomicseducation.org/data/mouse_autism_deseq_tissuetype_in_controlmice.csv)






