<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.26">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2026-02-23">

<title>Database Detectives: Exploring Public Genomic Databases Student Guide – GEMs</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-587c61ba64f3a5504c4d52d930310e48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-1db3f541d99483cdc7a0d9cf38862b2c.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://www.googletagmanager.com/gtag/js?id=G-XMR6CQ1L12"></script>

<script type="text/javascript">

window.dataLayer = window.dataLayer || [];
function gtag(){dataLayer.push(arguments);}
gtag('js', new Date());
gtag('config', 'G-XMR6CQ1L12', { 'anonymize_ip': true});
</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">GEMs</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../modules.html"> 
<span class="menu-text">Modules</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../resources.html"> 
<span class="menu-text">Resources</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../input.html"> 
<span class="menu-text">Share Your Input</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#the-suspected-genes" id="toc-the-suspected-genes" class="nav-link active" data-scroll-target="#the-suspected-genes">The suspected genes</a></li>
  <li><a href="#the-databases" id="toc-the-databases" class="nav-link" data-scroll-target="#the-databases">The databases</a></li>
  <li><a href="#part-1" id="toc-part-1" class="nav-link" data-scroll-target="#part-1">Part 1</a></li>
  <li><a href="#part-2" id="toc-part-2" class="nav-link" data-scroll-target="#part-2">Part 2</a></li>
  <li><a href="#part-3" id="toc-part-3" class="nav-link" data-scroll-target="#part-3">Part 3</a></li>
  <li><a href="#part-4" id="toc-part-4" class="nav-link" data-scroll-target="#part-4">Part 4</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Database Detectives: Exploring Public Genomic Databases Student Guide</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 23, 2026</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<p>You are a genetic detective! Your lab group has been running experiments for a variety of <em>Drosophila</em> genes. Unfortunately, there was an accident with a broken pipe and the computer, and all the stored data is gone! You have copies of the experimental results, but you haven’t been able to find the key that tells you which set of experiments corresponds to which gene. Your job is to find the background information on each gene using online databases, then match the fruit fly gene to the correct set of experimental results.</p>
<section id="the-suspected-genes" class="level3">
<h3 class="anchored" data-anchor-id="the-suspected-genes">The suspected genes</h3>
<p>Each mystery experiment is associated with one of four <em>Drosophila</em> genes: <strong>scarlet</strong>, <strong>plum</strong>, <strong>mustard</strong>, and <strong>white</strong>.</p>
<p>Each of these genes has been characterized in the <em>Drosophila</em> genome and you know their common name, but not much else. You’ll need to figure out where they’re located in the <em>Drosophila</em> genome, in what tissues they’re expressed, and what species share homologous genes. Finally, you’ll need to track down what possible human diseases their human homologs might be involved in.</p>
</section>
<section id="the-databases" class="level3">
<h3 class="anchored" data-anchor-id="the-databases">The databases</h3>
<p>For this activity, you’ll explore a variety of <em>curated genomic databases</em>. Many of these databases are available to the public at little or no cost and are supported by grants to the researchers and organizations that maintain them.</p>
<p><strong>Curated databases</strong> are simply databases where information is screened, validated, and organized by experts to ensure the data are accurate and reliable. Information has undergone rigorous quality control checks and includes sources and contextual information to help users understand the data. The data is generally standardized and organized in a way to be easily accessible and searchable.</p>
<p>It’s important to remember that the databases in this activity aren’t the only databases that exist! There are many more databases out there, each with its own focus. Luckily, most databases are organized in similar ways, so once you learn how to navigate one database, you’ll be able to easily figure out how to navigate new ones.</p>
<p>You should also keep in mind that some information can be found in multiple databases. For example, many databases that focus on the genome of a particular model organism will also have information about the human diseases associated with that particular gene. This information can also be found in databases that focus specifically on human genetic diseases (like OMIM or MalaCards). Just like there are many ways to cook an egg, there’s many ways to find the genomic information you might need in public databases. As long as you are careful and using a curated database, you can use the one you’re most comfortable with.</p>
</section>
<section id="part-1" class="level2">
<h2 class="anchored" data-anchor-id="part-1">Part 1</h2>
<p>Start with exploring FlyBase to get some basic information about each of the four genes.</p>
<p><a href="https://flybase.org/"><em>FlyBase</em></a> is a database focused on <em>Drosophila</em> genetics, genomics, and functional data. It is maintained by a consortium of Drosophila researchers and computer scientists at Harvard University, University of Cambridge (UK), Indiana University, and the University of New Mexico <span class="citation" data-cites="ozturk2024"><a href="#ref-ozturk2024" role="doc-biblioref">[1]</a></span>.</p>
<p>First, open the <a href="https://flybase.org/"><em>FlyBase</em></a> website by clicking the link or typing “https://flybase.org” into your browser. Then, type the name of the gene into the “J2G” bar in the upper right hand corner. (J2G stands for “Jump to Gene”.) As an example, try typing the gene name <strong>pipe</strong> and click enter.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g1965a5f7f0a_0_44.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>The gene page for <strong>pipe</strong> will load. On the top of the page, you will see the general information about this gene, including the <strong>FlyBase gene ID</strong> and a brief description of what this gene does.</p>
<p>Below this general information section, the genomic location data will appear. There are several different ways gene locations might be recorded; for our purposes, we’re interested in the <strong>sequence location</strong>. The <strong>pipe</strong> gene is located. “3L” means this gene is on the left arm of the third chromosome. The numbers after 3L refer to the nucleotide location.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g357ffc10808_0_3.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>You can scroll down the page or click the menu on the right-hand side to move to additional sections. The next section we’re interested in is the function section. Here, we can see what the general <strong>molecular function</strong> of the gene is, what sorts of <strong>biological processes</strong> the gene product (the protein) is involved in, and where the protein is found in the cell, or the <strong>cellular component</strong>. In the case of <strong>pipe</strong>, we can see that it’s an enzyme involved in development, gene expression, protein metabolism. We also see that the <strong>pipe</strong> protein is found in the membrane and endomembrane system, as well as in the nucleus around the chromosomes.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g357ffc10808_0_12.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Scrolling down further takes us to a section with more details about the gene function. You may need to click on the small plus sign to view the details (or click on the minus sign to hide the details). We can see that <strong>pipe</strong> codes for an enzyme that is enables endopeptidase activator activity and protein binding.</p>
<p>Now it’s your turn to look up your four genes in FlyBase.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>TASK ONE
</div>
</div>
<div class="callout-body-container callout-body">
<p>For each gene (<strong>scarlet</strong>, <strong>plum</strong>, <strong>mustard</strong>, and <strong>white</strong>), find the following information:</p>
<ul>
<li>FlyBase gene ID</li>
<li>chromosomal location</li>
<li>molecular function</li>
<li>the biological processes the gene product is involved in</li>
<li>where in the cell the gene product can be found</li>
</ul>
</div>
</div>
</section>
<section id="part-2" class="level2">
<h2 class="anchored" data-anchor-id="part-2">Part 2</h2>
<p>Next, let’s explore additional genomic databases beyond FlyBase to characterize the homologs and orthologs to the four <em>Drosophila</em> genes.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>A <strong>homolog</strong> is a gene or protein that is similar to another due to having a common evolutionary origin. Many genes in <em>Drosophila</em> have homologs in the human genome, as well as in the genomes of common model organisms like mice and zebrafish.</p>
<p>An <strong>ortholog</strong> is another term for a gene or protein that is similar due to having a common evolutionary origin, but in order to be an ortholog, the two genes must <em>also</em> have similar functions within the different species. An ortholog is a specialized kind of homolog.</p>
<p>Genomic databases will report both homologs and orthologs.</p>
</div>
</div>
<p>First, we need to find out what the name of the homologous genes are. The gene IDs and common names differ between species. If we keep scrolling down the page on the <strong>pipe</strong> gene in FlyBase, we can find what those homologous genes are called in our species of interest.</p>
<p>It turns out that the human gene most likely to be a homolog of <strong>pipe</strong> is called <strong>UST</strong>.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g357ffc10808_0_30.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>If we scroll down a little farther, we see that the most likely homologous mouse (<em>Mus musculus</em>) is called <strong>Ust</strong>. In the zebrafish (<em>Danio rerio</em>), it is most likely to be <strong>usta</strong>.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g357ffc10808_0_61.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>To find out more information about these homologs, we’re going to need to look at additional species-specific databases. Let’s start with the mouse.</p>
<p>The <a href="https://informatics.jax.org/"><em>Mouse Genome Informatics</em></a> database is the international database for the laboratory mouse, integrating genetic, genomic, and biological data as part of an effort to use the mouse to study human health and disease. It is maintained by the Jackson Laboratory <span class="citation" data-cites="baldarelli2024"><a href="#ref-baldarelli2024" role="doc-biblioref">[2]</a></span>.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Why do we have a database devoted to the mouse?</p>
<p>The mouse is the most commonly-used model organism in laboratory work. In fact, mice and rats make up 95% of the lab animal population, and more than 80% of the research that has been awarded the Nobel Prize for Medicine was done at least in part with mouse models (https://www.cshl.edu/of-mice-and-model-organisms/, https://fbresearch.org/medical-advances/nobel-prizes).</p>
<p>So what makes mice such good model organisms for biomedical research? One reason is that mice and humans are both mammals and have about 85% of their protein-coding genome in common. As a result, mouse physiology is quite similar to human physiology. The mouse circulatory, reproductive, digestive, hormonal, and nervous systems are frequently used as models to study how humans grow, age, and develop chronic diseases.</p>
</div>
</div>
<p>Open the MGI database and search for the name of the mouse homolog <strong>Ust</strong>. You can do this either by typing <strong>Ust</strong> into the search bar and clicking “Quick Search” or by doing a topic specific search with the “Genes” toolbar.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35f41619954_0_41.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>A “Genome Features” summary table should open up with information about <strong>Ust</strong>. MGI tells us that <strong>Ust</strong> is a protein-coding gene on chromosome 10. Clicking on the highlighted gene symbol will take us to a page with more detail.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35f41619954_0_46.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>The kinds of information you found in FlyBase will also be present on the detailed gene page in MGI. The information is organized into sections, similar to what you’ve already experienced with FlyBase. The fourth section down includes information about the human ortholog to <strong>Ust</strong> - it turns out to be <strong>UST</strong>, which we already knew from FlyBase.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35e6d9f15b7_13_34.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Farther down the page is a section called “Gene Ontology (GO) Classifications”. This is the section that has information about the molecular functions, biological processes, and cellular components for <strong>Ust</strong>.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>“Gene ontology” is a bioinformatics initiative focused on standardizing the vocabulary and annotations researchers use to describe genes, proteins, and data about gene and protein function. This effort allows us to more easily make comparisons about homologous and orthologous genes and proteins across species.</p>
</div>
</div>
<p>In mice, <strong>Ust</strong> is a tranferase (an enzyme!) involved in carbohydrate derivative binding.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35e6d9f15b7_13_40.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>In order to look up information about the zebrafish homolog, we need to switch databases again.</p>
<p>The <a href="https://zfin.org/"><em>ZFIN (Zebrafish Information Network </em></a> is the database of genetic and genomic data for the zebrafish as a model organism. It was formed by a group of zebrafish researchers at Cold Spring Harbor and is maintained by the ZFIN Database Team at the University of Oregon <span class="citation" data-cites="bradford2022"><a href="#ref-bradford2022" role="doc-biblioref">[3]</a></span>.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Why do we have a database devoted to the zebrafish?</p>
<p>Zebrafish are a popular model organism for researchers who focus on developmental biology, genetics, and modeling diseases. They develop rapidly and have high fertility, so it can be relatively quick and fast to study the effects of genes or diseases over multiple generations. They are also transparent, which is especially useful for watching how organs develop as the zebrafish matures.</p>
<p>Zebrafish share a high degree of genetic homology with humans!</p>
</div>
</div>
<p>Open ZFIN and search for the zebrafish homolog <strong>usta</strong>. The ZFIN search bar will offer some options to refine your search. Choose “Gene/Transcript” from the drop down menu.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35e6d9f15b7_13_13.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>ZFIN takes you directly to the gene page for <strong>usta</strong>. A summary of basic information about <strong>usta</strong> is at the top, with more detailed sections as you scroll down. Without much searching, we discover that <strong>usta</strong> is located on chromosome 20 and is predicted to enable sulfotransferase activity.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35e6d9f15b7_13_8.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Clicking on “Gene Ontology” in the menu on the left takes you to a more detailed list of the molecular functions associated with <strong>usta</strong>. The summary on top just mentioned the words “sulfotransferase activity”; in the Gene Ontology section, we learn that <strong>usta</strong> is involved in catalytic activity and enzyme regulator activity as well.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35e6d9f15b7_13_51.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Finally, we can turn to the human homolog <strong>UST</strong>. There are several databases with information about human genes. <a href="https://www.genecards.org/"><em>GeneCards</em></a> is a user-friendly database that collects and displays information about human genes from multiple research databases. It is maintained as a joint effort between the Weizmann Institute of Science and LifeMap Sciences <span class="citation" data-cites="stelzer2016"><a href="#ref-stelzer2016" role="doc-biblioref">[4]</a></span>.</p>
<p>Open the GeneCards page and search for the human homolog <strong>UST</strong> in the “Explore a Gene” search bar.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35e6d9f15b7_13_22.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Unlike MGI or ZFIN, the GeneCards page on our searched gene doesn’t have the information about location at the top. Instead, we will need to click on “Genomics” in the header menu.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35e6d9f15b7_13_64.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>The genomics section of the GeneCards page includes location information for multiple assemblies, as well as a map of the chromosome. <strong>UST</strong> is located on chromosome 6.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>A genome assembly is just a version of the genome map. As scientists unlock more detail about genomes, they periodically publish an updated “reference genome”, with everything we have discovered about where genes, enhancers, promotors, and non-coding regions are located in the genome. Think of an updated genome assembly as a better map.</p>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35e6d9f15b7_13_73.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>We can also find information about <strong>UST’s</strong> molecular function. Choosing “Function” from the menu takes us to a list of gene ontology terms (commonly called GO terms) associated with <strong>UST</strong>. There are a lot of words you might recognize, including sulfotransferase activity.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35e6d9f15b7_13_27.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Now it’s your turn to look up information about the homologs of your four <em>Drosophila</em> genes.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>TASK TWO
</div>
</div>
<div class="callout-body-container callout-body">
<p>For each of the <em>Drosophila</em> genes (<strong>scarlet</strong>, <strong>plum</strong>, <strong>mustard</strong>, and <strong>white</strong>), find the following information for the <em>mouse, zebrafish, and human homologs</em>:</p>
<ul>
<li>name</li>
<li>chromosomal location</li>
<li>molecular function</li>
</ul>
<p>Does this information differ among the homologs? For example, do the homologs of <strong>scarlet</strong> do the same things as <strong>scarlet</strong>?</p>
</div>
</div>
</section>
<section id="part-3" class="level2">
<h2 class="anchored" data-anchor-id="part-3">Part 3</h2>
<p>In addition to the experimental data you’re sorting through, you notice that someone has collected literature on different human diseases. You assume these diseases are thought to be caused in part by some of the human homologs of the genes your lab is studying, so you decide to look up the information using a pair of human disease databases.</p>
<p><a href="https://omim.org/"><em>OMIM</em></a>, short for Online Mendelian Inheritance in Man, is a comprehensive database of human genes and genetic phenotypes. It is maintained by the McKusick-Nathans Institute of Genetic Medicine at Johns Hopkins University.</p>
<p><a href="https://www.malacards.org/"><em>MalaCards</em></a> is a searchable database offering information on human diseases, medical conditions, and disorders. It is maintained as a joint effort between the Weizmann Institute of Science and LifeMap Sciences.</p>
<p>Both databases will give you the information you need. We’ll go through how to use both, but you only need to use one for this activity.</p>
<p>Let’s start with <a href="https://omim.org/"><em>OMIM</em></a>. Click on the link or type “https://omim.org” into your browser. On the landing page, you can type either a disease or a gene name into the search bar. (Just remember to use the human gene name or gene ID!)</p>
<p>Earlier, we discovered the human homolog of <strong>pipe</strong> is <strong>UST</strong>. Let’s look this gene up and see what (if any) diseases it might be associated with.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35f41619954_0_23.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Our search pulls up a list of entries associated with our search term. For <strong>UST</strong>, there’s only one. Click on it.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35f41619954_0_36.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Each entry in OMIM will have a table of contents on the left hand side of the page. In the case of <strong>UST</strong>, there’s not much there. In fact, there doesn’t seem to be any subsection focusing on disease state or phenotype.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35f41619954_0_51.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>As a comparison, we can search for the known disease risk gene <strong>BRCA1</strong>, which is associated with an increased risk of breast and ovarian cancer. When we look at the top OMIM hit for <strong>BRCA1</strong>, we can see a section in the table of contents called “Gene-Phenotype Relationships”. This is the section where OMIM displays any possible disease phenotype associated with the gene.</p>
<p>Since we don’t see this section for <strong>UST</strong>, we can conclude that <strong>UST</strong> doesn’t appear to cause any human diseases that we know of at this time.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g3367b17b963_0_3.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>You can find similar information on the <a href="https://www.malacards.org/"><em>MalaCards</em></a> website. Click the link or type “https://www.malacards.org” into your browser to open the webpage. Then type “UST” into the bottom search bar. (Although you can look up genes on MalaCards, most of the time you look up a a disease to find out what genes are associated with it instead of the other way around.)</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35f41619954_0_55.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Since we used the “search any term” search bar, MalaCards will look for the term “UST” on all the disease pages in the database. The results are ordered with the best-matching page on top. Click “Crohn’s Disease” to see why “UST” matched this entry.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g35f41619954_0_59.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>One nice thing about MalaCards is that the search term is highlighted in yellow when you open the page. In this case, we can see that “UST” actually matched part of drug that was investigated as a treatment for Crohn’s disease. This suggests that the gene <strong>UST</strong> is not found in any of the MalaCards disease entry.</p>
<p>Let’s search MalaCards with the term “BRCA1” to see what it looks like when the gene name <em>is</em> found in an entry. In the top hit entry (“Hereditary Breast Ovarian Cancer Syndrome), we can see the actual gene name <strong>BRCA1</strong> shows up in the overview, as well as in the section about associated genes and the alternate name for the disease.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="database_detectives_student_guide_files/figure-html//1fy3FwkSLETnRUdf8huddq2n4qcb6eOyzwVg05dbudII_g3367b17b963_0_13.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%" alt="TO ADD"></p>
</figure>
</div>
</div>
</div>
<p>Now it’s your turn to look up information the diseases (if any) your four human homologs might be associated with.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>TASK THREE
</div>
</div>
<div class="callout-body-container callout-body">
<p>What human disease or disorder (if any) are the homologs of <strong>scarlet</strong>, <strong>plum</strong>, <strong>mustard</strong>, and <strong>white</strong> associated with?</p>
</div>
</div>
</section>
<section id="part-4" class="level2">
<h2 class="anchored" data-anchor-id="part-4">Part 4</h2>
<p>Finally, it’s time for you to report back to your supervisor. Here’s what you know about the experiments:</p>
<p><strong>Experiment 1</strong>: the data seems to be from the 3R chromosome in <em>Drosophila</em> and measures whether motor neurons are being pruned at the same rate in the experimental group (versus the control group)</p>
<p><strong>Experiment 2</strong>: this data includes measurements on transmembrane transport in the zebrafish, specifically in intestinal cells</p>
<p><strong>Experiment 3</strong>: this data looks at mouse cells in culture and tracks the damage caused by exposure to reactive oxygen species (ROS) and oxidative stress</p>
<p><strong>Experiment 4</strong>: this data includes sequences from the <em>Drosophila</em> X chromosome and appears to be a genome annotation project</p>
<p>There is also extensive background research about cerebellar atrophy, gallbladder disease, and hyperuricemia (high uric acid concentration in the urine). You also find some initial research looking at neuronal cell adhesion.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>TASK FOUR
</div>
</div>
<div class="callout-body-container callout-body">
<ol type="1">
<li><p>Which <em>Drosophila</em> genes are associated with each experiment?</p></li>
<li><p>Which background research on human disease belongs with each experiment?</p></li>
</ol>
</div>
</div>



</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body" data-entry-spacing="0" role="list">
<div id="ref-ozturk2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">[1] </div><div class="csl-right-inline">A. Öztürk-Çolak <em>et al.</em>, <span>“FlyBase: Updates to the drosophila genes and genomes database,”</span> <em>Genetics</em>, vol. 227, no. 1, Feb. 2024, Available: <a href="https://doi.org/10.1093/genetics/iyad211">https://doi.org/10.1093/genetics/iyad211</a></div>
</div>
<div id="ref-baldarelli2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">[2] </div><div class="csl-right-inline">R. M. Baldarelli, C. L. Smith, M. Ringwald, J. E. Richardson, C. J. Bult, and M. G. I. Group, <span>“Mouse genome informatics: An integrated knowledgebase system for the laboratory mouse,”</span> <em>Genetics</em>, vol. 227, no. 1, Mar. 2024, Available: <a href="https://doi.org/10.1093/genetics/iyae031">https://doi.org/10.1093/genetics/iyae031</a></div>
</div>
<div id="ref-bradford2022" class="csl-entry" role="listitem">
<div class="csl-left-margin">[3] </div><div class="csl-right-inline">Y. M. Bradford <em>et al.</em>, <span>“Zebrafish information network, the knowledgebase for danio rerio research,”</span> <em>Genetics</em>, vol. 220, no. 4, Feb. 2022, Available: <a href="https://doi.org/10.1093/genetics/iyac016">https://doi.org/10.1093/genetics/iyac016</a></div>
</div>
<div id="ref-stelzer2016" class="csl-entry" role="listitem">
<div class="csl-left-margin">[4] </div><div class="csl-right-inline">G. Stelzer <em>et al.</em>, <span>“The GeneCards suite: From gene data mining to disease genome sequence analyses,”</span> <em>Current Protocols in Bioinformatics</em>, vol. 54, no. 1, pp. 1.30.1–1.30.33, 2016, Available: <a href="https://currentprotocols.onlinelibrary.wiley.com/doi/abs/10.1002/cpbi.5">https://currentprotocols.onlinelibrary.wiley.com/doi/abs/10.1002/cpbi.5</a></div>
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/genomicseducation\.org");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
            // target, if specified
            link.setAttribute("target", "_blank");
            if (link.getAttribute("rel") === null) {
              link.setAttribute("rel", "noopener");
            }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p style="text-align:center;">
<small> This work is licensed under a <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License</a>. </small>
</p>
<p style="text-align:center;">
<small> GEMs is funded by the National Human Genome Research Institute <a href="https://reporter.nih.gov/search/SjWuEKrSVEGVaHvRb9L-vg/project-details/10854368">1R25HG013489.</a>
</small></p><small>
</small><p><small></small></p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>